src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -130,6 +146,8 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
                 telemetry.TelemetryEvent[HttpSourcesCount] = httpSourcesCount;
                 telemetry.TelemetryEvent[LocalSourcesCount] = _request.DependencyProviders.RemoteProviders.Count - httpSourcesCount;
                 telemetry.TelemetryEvent[FallbackFoldersCount] = _request.DependencyProviders.FallbackPackageFolders.Count;
+                telemetry.TelemetryEvent[ProjectHasFloatingVersions] = _request.Project.TargetFrameworks.SelectMany(tfm => tfm.Dependencies)
+                    .Where(d => d.LibraryRange?.VersionRange != null && IsOriginalStringVersionRangeOrFloating(d.LibraryRange.VersionRange)).Any();
```

Comment:
I understand to an extent why we'd benefit from knowing the floating version usage, but how does the version range notation help us?

Reply:
In the beginning I added this to measure the usage of the feature, but since the Telemetry wont be used like that now (it is not an experiment now) you think is goo to have separated Telemetry for Ranges or just keep the Floating?

Reply:
I see. 

So to understand feature usage, this is probably too late correct? 

Should we only be adding telemetry for things that we know we'll use such as the how often floating versions are entered through the PM UI? 

The thing that I'd like to avoid is the string manipulation, such as startswith `[` because imo it doesn't tell us much.

Is there anything important we'd learn by having telemetry abouyt ranges/floating versions from the project level event?

Reply:
What I'm trying to say is that the information that the PM Ui is being used for adding floating versions can be sent in the nugetaction event. 

When you get to the vs/nuget/restoreinformation, you don't really need that information anymore, because you can't know  how exactly it got added.

Reply:
That makes sense in theory I guess, but there's not *much* to this feature right now. It's fairly complete. 

I'm ok if we really want to add details about floating version usage. I don't think we need info about ranges, as I don't see a use case for that.

Reply:
If you do decide to keep it, please deconstruct the linq and use an early exit once a floating version is seen.

Reply:
@nkolev92 I changed it to floating versions only. Is `Any(condtion)` enough? I think it will return quickly once it founds a floating version.